test/compress/collapse_vars.js

```diff
@@ -68,11 +68,10 @@ collapse_vars_side_effects_1: {
             log(x, s.charAt(i++), y, 7);
         }
         function f4() {
-            var log = console.log.bind(console),
-                i = 10,
+            var i = 10,
                 x = i += 2,
                 y = i += 3;
-            log(x, i += 4, y, i);
+            console.log.bind(console)(x, i += 4, y, i);
```

Comment:
While you and I can deduce this optimization is correct, how can `collapse_vars` know that the side effects in `console.log.bind` do not interact with the modification of `i` above?